### ISSUE TITLE
DOMException: Failed to execute 'postMessage' on 'MessagePort': ArrayBuffer at index 0 is already detached.

### ISSUE THREAD

**Issue by Nek** [Tonejs/Tone.js#915]
Tonejs throws an error while trying to load a sample with devtools open.

**To Reproduce**

1. Go to this code sandbox https://codesandbox.io/s/gracious-mountain-q8in2?file=/src/index.ts
2. Open the devtools in Chrome
3. Reload the app preview

Observe an error being thrown.

**Expected behavior**
It shouldn't throw and should just load and play a sound.

**Comment by tambien** [comment]
I'm not seeing this error in the console. Is there supposed to be Tone.js code in index.ts?

**Comment by Nek** [comment]
I'm sorry. It looks like the draft didn't get saved. Here's the updated version (the same link, actually):
https://codesandbox.io/s/gracious-mountain-q8in2?file=/src/index.ts

**Comment by tambien** [comment]
I am unable to reproduce this issue on Chrome `Version 92.0.4515.107 (Official Build) (x86_64)`

When i refresh the preview pane, this is what my console looks like: 

![image](https://user-images.githubusercontent.com/1323402/126905325-c4d4ab35-dad5-439e-b5f2-b9b5dacaef32.png)

The error that's thrown, which file does it point to in Tone.js?

**Comment by Nek** [comment]
It's my bad again, one have to click the sandbox preview to startup sample fetching and observe the error.
The error happens in ToneAudioBuffer. The actual throw site is detach-array-buffer.js:14 of standardized-audio-context.

**Comment by tambien** [comment]
Unfortunately still unable to get this error. Whenever i click on the preview pane, i hear the chord. 

Which version of Chrome are you using?

**Comment by chrisguttandin** [comment]
Hi everyone, I'm not sure why this sometimes pops up in Chrome and sometimes doesn't, I assume it's a race condition. I saw it in one of my apps today and was able to reproduce it.

The problem was that Safari now returns a promise as well from `decodeAudioData()`. Unfortunately it still doesn't detach the `ArrayBuffer` as defined in the spec. Therefore standardized-audio-context does that manually. But this causes problems in Chrome (which executes the same code path for browsers with promise support) since Chrome detaches the `ArrayBuffer` itself and detaching it again throws an error.

It should be fixed now with v25.3.3 of standardized-audio-context. Here is the commit with the fix: https://github.com/chrisguttandin/standardized-audio-context/commit/e5ce31d95d48ad0a2368b146f4b9100f8e698484

**Comment by Nek** [comment]
Thanks for the fix!  I have started to question my sanity at some point. :)

**Comment by tambien** [comment]
thanks @chrisguttandin! super helpful. i will update `standard-audio-context` in the next build

**Comment by 80prozent** [comment]
@chrisguttandin yes and no :). my change just catches the exception, so chrome-dev-tools no longer pauses on these exceptions.
the `finally` statement will still execute the same. the only difference is that chrome-dev-tools stops complaining

**Comment by OndrejSpanel** [comment]
I am getting this error today in Chrome. I am working with minified version from https://cdnjs.cloudflare.com/ajax/libs/tone/14.8.49/Tone.min.js. The minified code is:

```js
i.onmessage=()=>e();try{n.postMessage(s,[s])}finally{e()}
```

I can see the `catch` statement in the [master of the file in question](https://github.com/chrisguttandin/standardized-audio-context/blob/master/src/helpers/detach-array-buffer.ts), therefore I does not seem the fix was removed but perhaps something in the build process (Typescript compilation, or more likely minification) makes the fix disappear in minified code?

Not having the `catch` present makes debugging application using Tone.js in Chrome unreliable and frustrating.
